### PR TITLE
SoftPlus operation specification

### PR DESCRIPTION
**Pull request by achetver**
Description: Add specification for SoftPlus activation operation.

JIRA: CVS-33002

### CODE REVIEW

**Comment by achetver** [comment]
@lazarevevgeny Should we add new file `docs/ops/opset4.md`?

File: docs/ops/activation/SoftPlus_4.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## SoftPlus <a name="SoftPlus"></a>
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:1]: @dmitry-gorokhov, I see that in MKLDNN plugin the operation is called "Softplus" (lower case "p"). Could you adjust the name to the specification? MO never emitted IR with operation "Softplus" so the change will not break backward compatibility.

File: docs/ops/activation/SoftPlus_4.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+*   **1**: Multidimensional input tensor of type *T*. **Required**.
+
+**Outputs**:
+
+*   **1**: The resulting tensor of the same shape and type as input tensor.
+
+**Types**
+
+* *T*: any supported type.
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:25]: I think that we want to support only floating point types (not integer and boolean).

File: docs/ops/activation/SoftPlus_4.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+## SoftPlus <a name="SoftPlus"></a>
+
+**Versioned name**: *SoftPlus-4*
+
+**Category**: *Activation*
+
+**Short description**: SoftPlus takes one input tensor and produces output tensor where the softplus function is applied to the tensor elementwise.
+
+**Detailed description**: For each element from the input tensor calculates corresponding
+element in the output tensor with the following formula: `features[i] = log(exp(features[i]) + 1)`
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:10]: Please, change "1" to "1.0"

File: docs/ops/activation/SoftPlus_4.md (hunk 2 above)
Comment (achetver) [reply docs/ops/activation/SoftPlus_4.md]: Ok, updated.

File: docs/ops/activation/SoftPlus_4.md (hunk 3 above)
Comment (achetver) [reply docs/ops/activation/SoftPlus_4.md]: Done.

File: docs/ops/activation/SoftPlus_4.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+## SoftPlus <a name="SoftPlus"></a>
+
+**Versioned name**: *SoftPlus-4*
+
+**Category**: *Activation*
+
+**Short description**: SoftPlus takes one input tensor and produces output tensor where the softplus function is applied to the tensor elementwise.
+
+**Detailed description**: For each element from the input tensor calculates corresponding
+element in the output tensor with the following formula: `features[i] = log(exp(features[i]) + 1.0)`
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:10]: Rewrite the mathematical function like in https://github.com/openvinotoolkit/openvino/pull/1149

Comment (achetver) [reply docs/ops/activation/SoftPlus_4.md]: Done.

File: docs/ops/activation/SoftPlus_4.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+**Category**: *Activation*
+
+**Short description**: SoftPlus takes one input tensor and produces output tensor where the softplus function is applied to the tensor elementwise.
+
+**Detailed description**: For each element from the input tensor calculates corresponding
+element in the output tensor with the following formula:
+
+    \f[
+    SoftPlus( x ) = log{e^{x} + 1.0}
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:13]: Please, specify what is the base value for `log`. Should it be `ln`?

Comment (achetver) [reply docs/ops/activation/SoftPlus_4.md]: Yes? it should be `ln`. Specification updated.

File: docs/ops/activation/SoftPlus_4.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+**Category**: *Activation*
+
+**Short description**: SoftPlus takes one input tensor and produces output tensor where the softplus function is applied to the tensor elementwise.
+
+**Detailed description**: For each element from the input tensor calculates corresponding
+element in the output tensor with the following formula:
+
+    \f[
+    SoftPlus( x ) = ln{e^{x} + 1.0}
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:13]: Use "()" instead of "{}" to surround content of `ln` function.

Comment (achetver) [reply docs/ops/activation/SoftPlus_4.md]: Done.

**Pull request merged**